### ISSUE TITLE
[Rule Request] Allow for (let i ...) loops with prefer-for-of when the iterated variable is a string

### ISSUE THREAD

**Issue by JakeTunaley** [palantir/tslint#4006]
- __TSLint version__: 5.10.0
- __TypeScript version__: 3.0.0-dev.20180628
- __Running TSLint via__: VSCode

### Issue

```ts
const str = "hello world";
for (let i = 0; i < str.length; i++) { // [tslint] Expected a 'for-of' loop instead of a 'for' loop with this simple iteration (prefer-for-of)
	console.log(str[i]);
}
for (const chr of str) { // [ts] Using a string in a 'for...of' statement is only supported in ECMAScript 5 and higher.
	console.log(chr);
}
```

This is annoying because it means I have to put `tslint:disable-next-line` rules everywhere I iterate over a string.

### Suggestion

Add an "ignore-strings" (or similarly named) option to the "prefer-for-of" rule so that I can keep the rule on without having to put disable comments everywhere.

### tslint.json

```json
{
    "defaultSeverity": "error",
    "extends": [
        "tslint:recommended"
    ],
    "jsRules": {},
    "rules": {
        "no-console": false,
        "indent": [true, "tabs"],
        "max-line-length": false
    },
    "rulesDirectory": []
}
```

### tsconfig.json

```json
{
	"compilerOptions": {
		"target": "es3"
	}
}
```

**Comment by giladgray** [comment]
not into a type-based rule as it will only be available to users who use the `--project` option. if the rule does not work for a common use case in your codebase then you can disable the rule to allow such syntax.


there was a recent issue about enabling this `for (let i)` syntax but I can't find it right now.

**Comment by ajafff** [comment]
@JakeTunaley If you want `prefer-for-of` to use type information to avoid false positives, you might want to give my [Fimbullinter project](https://github.com/fimbullinter/wotan/blob/master/packages/wotan/README.md) a try. Like TSLint it has a rule called [`prefer-for-of` ](https://github.com/fimbullinter/wotan/blob/master/packages/mimir/README.md#rules) which suggests converting regular `for` loops to `for-of` loops, but avoids false positives:

* when targeting ES3 iterated value must not be a string
* when targeting ES5 or below without `--downlevelIteration`: iterated value needs to be a string or extend `Array` or `ReadonlyArray`
* when targeting ES2015 or higher or using `--downlevelIteration`: iterated value needs to implement the iteration protocol and the iterator must yield values of the same type as the index signature

**Comment by JakeTunaley** [comment]
@ajafff Thanks, that's exactly what I need!

**Comment by johnwiseheart** [comment]
This isn't worth the cost of implementation of a type-based rule, especially since they're only supported by the CLI anyway
